backend/tracim_backend/tests/functional/test_contents.py

```diff
@@ -29,6 +29,26 @@ class TestFolder(FunctionalTest):
 
     fixtures = [BaseFixture]
 
+    def _setup_basics(self) -> None:
+        self.testapp.authorization = ("Basic", ("admin@admin.admin", "admin@admin.admin"))
+        self.dbsession = get_tm_session(self.session_factory, transaction.manager)
+        self.admin = self.dbsession.query(User).filter(User.email == "admin@admin.admin").one()
+        self.workspace_api = WorkspaceApi(
+            current_user=self.admin, session=self.dbsession, config=self.app_config
+        )
+        self.content_api = ContentApi(
+            current_user=self.admin, session=self.dbsession, config=self.app_config
+        )
+        self.workspace = self.workspace_api.create_workspace(label="test", save_now=True)
+        self.folder = self.content_api.create(
```

Comment:
this data is part of the test case, not the context, so it should move into the test body itself. otherwise, the reading of the test case `test_api__move_folder_into_itself__err_400` is not clear: We do not understand what `self.folder` is.

setup() and tearDown() are about context, all data directly included in the test must be created in the test.